pkg/resources/streamlit.go

```diff
@@ -0,0 +1,408 @@
+package resources
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/helpers"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/logging"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/provider"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/schemas"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/customdiff"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var streamlitSchema = map[string]*schema.Schema{
+	"name": {
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "String that specifies the identifier (i.e. name) for the streamlit; must be unique in your account.",
+	},
+	"database": {
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "The database in which to create the Cortex search service.",
+		ForceNew:    true,
+	},
+	"schema": {
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "The schema in which to create the streamlit.",
+		ForceNew:    true,
+	},
+	"stage": {
+		Type:             schema.TypeString,
+		Required:         true,
+		Description:      "The stage in which streamlit files are located.",
+		ValidateDiagFunc: IsValidIdentifier[sdk.SchemaObjectIdentifier](),
+		DiffSuppressFunc: SuppressIfAny(suppressIdentifierQuoting, IgnoreChangeToCurrentSnowflakeValueInOutput(DescribeOutputAttributeName, "root_location")),
+	},
+	"directory_location": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		Description:      "Specifies the full path to the named stage containing the Streamlit Python files, media files, and the environment.yml file.",
+		DiffSuppressFunc: IgnoreChangeToCurrentSnowflakeValueInOutput(DescribeOutputAttributeName, "root_location"),
+	},
+	"main_file": {
+		Type:             schema.TypeString,
+		Required:         true,
+		Description:      "Specifies the filename of the Streamlit Python application. This filename is relative to the value of `root_location`",
+		DiffSuppressFunc: IgnoreChangeToCurrentSnowflakeValueInOutput(DescribeOutputAttributeName, "main_file"),
+	},
+	"query_warehouse": {
+		Type:             schema.TypeString,
+		ValidateDiagFunc: IsValidIdentifier[sdk.AccountObjectIdentifier](),
+		Optional:         true,
+		Description:      "Specifies the warehouse where SQL queries issued by the Streamlit application are run.",
+		DiffSuppressFunc: SuppressIfAny(suppressIdentifierQuoting, IgnoreChangeToCurrentSnowflakeValueInOutput(ShowOutputAttributeName, "query_warehouse")),
+	},
+	"external_access_integrations": {
+		Type: schema.TypeSet,
+		Elem: &schema.Schema{
+			Type: schema.TypeString,
+		},
+		Optional:         true,
```

Comment:
This should have diff suppression added in network policy to avoid infinite plans that can occur with quoting differences like in (https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/2236#issuecomment-2220003509).